packages/core/components/ExternalInput/index.tsx

```diff
@@ -100,13 +113,44 @@ export const ExternalInput = ({
           className={getClassNameModal({
             isLoading,
             loaded: !isLoading,
-            hasData: !!data,
+            hasData: data.length > 0,
           })}
         >
           <div className={getClassNameModal("masthead")}>
             <Heading rank={2} size="xxl">
               Select content
             </Heading>
+
+            {field.showSearch && (
+              <form
+                className={getClassNameModal("searchForm")}
+                onSubmit={(e) => {
+                  e.preventDefault();
+
+                  search(searchQuery);
+                }}
+              >
+                <label className={getClassNameModal("search")}>
+                  <span style={{ visibility: "hidden", width: 0 }}>Search</span>
+                  <div className={getClassNameModal("searchIcon")}>
+                    <Search size="18" />
+                  </div>
+                  <input
```

Comment:
Could maybe use `type="search"` (and follow the convention of `name=q`), see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search